microprofile/openapi/src/main/java/io/helidon/microprofile/openapi/Serializer.java

```diff
@@ -221,31 +254,50 @@ private void processExtensions(MappingNode node, Object javaBean) {
                 return;
             }
 
-            List<NodeTuple> tuples = new ArrayList<>(node.getValue());
+            List<NodeTuple> tuples = node.getValue();
 
             if (tuples.isEmpty()) {
                 return;
             }
+            List<NodeTuple> updatedTuples = processExtensions(tuples);
+            node.setValue(updatedTuples);
+        }
+
+        /**
+         * Returns a (possibly) updated list of {@code NodeTuple} objects, promoting any node under the node {@code extensions}
+         * one level higher and removing the {@code extensions} node itself.
+         *
+         * @param tuples {@code NodeTuple} objects to process
+         * @return possibly updated list of node tuples
+         */
+        private List<NodeTuple> processExtensions(List<NodeTuple> tuples) {
             List<NodeTuple> updatedTuples = new ArrayList<>();
 
             tuples.forEach(tuple -> {
-                Node keyNode = tuple.getKeyNode();
-                if (keyNode.getTag().equals(Tag.STR)) {
-                    String key = ((ScalarNode) keyNode).getValue();
-                    if (key.equals(EXTENSIONS)) {
-                        Node valueNode = tuple.getValueNode();
-                        if (valueNode.getNodeId().equals(NodeId.mapping)) {
-                            MappingNode extensions = MappingNode.class.cast(valueNode);
-                            updatedTuples.addAll(extensions.getValue());
-                        }
-                    } else {
-                        updatedTuples.add(tuple);
-                    }
-                } else {
+                // Returns either null (if the tuple is not for "extensions") or a list of tuples for each extension property.
+                List<NodeTuple> extensionTuples = processExtensions(tuple);
+                if (extensionTuples == null) {
                     updatedTuples.add(tuple);
+                } else {
+                    updatedTuples.addAll(extensionTuples);
                 }
             });
-            node.setValue(updatedTuples);
+            return updatedTuples;
+        }
+
+        private List<NodeTuple> processExtensions(NodeTuple tuple) {
+            Node keyNode = tuple.getKeyNode();
+            if (keyNode.getTag().equals(Tag.STR)) {
+                String key = ((ScalarNode) keyNode).getValue();
+                if (key.equals(EXTENSIONS)) {
+                    Node valueNode = tuple.getValueNode();
+                    if (valueNode.getNodeId().equals(NodeId.mapping)) {
```

Comment:
Maybe:
```
if (keyNode.getTag().equals(Tag.STR)
     && ((ScalarNode) keyNode).getValue().equals(EXTENSIONS)
     && tuple.getValueNode() instanceof MappingNode extensions
     && extensions.getNodeId().equals(NodeId.mapping)) {
    return extensions.getValue();
}
```
…if I've typed it all right. Saves some nesting and makes it a little lighter to read.

Reply:
Changed